plugins/bufferlines/barbar.nix

```diff
@@ -69,249 +24,567 @@ let
     '';
 
-    insertAtStart = helpers.defaultNullOpts.mkBool false ''
-      If true, new buffers will be inserted at the start of the list.
-      Default is to insert after current buffer.
+    highlight_inactive_file_icons = helpers.defaultNullOpts.mkBool false ''
+      Enables highlighting the file icons of inactive buffers.
     '';
 
-    maximumPadding =
-      helpers.defaultNullOpts.mkInt 4
-        "Sets the maximum padding width with which to surround each tab";
+    highlight_visible = helpers.defaultNullOpts.mkBool true ''
+      Enables highlighting of visible buffers.
+    '';
+
+    icons = {
+      buffer_index =
+        helpers.defaultNullOpts.mkNullable
+          (
+            with types;
+            either bool (enum [
+              "superscript"
+              "subscript"
+            ])
+          )
+          false
+          "If `true`, show the index of the buffer with respect to the ordering of the buffers in the tabline.";
+
+      buffer_number = helpers.defaultNullOpts.mkNullable (
+        with types;
+        either bool (enum [
+          "superscript"
+          "subscript"
+        ])
+      ) false "If `true`, show the `bufnr` for the associated buffer.";
+
+      button = helpers.defaultNullOpts.mkNullable (with types; either str (enum [ false ])) "" ''
+        The button which is clicked to close / save a buffer, or indicate that it is pinned.
+        Use `false` to disable it.
+      '';
+
+      diagnostics = mkOption {
+        type = types.submodule {
+          options =
+            let
+              mkDiagnosticIconOptions = iconDefault: {
+                enabled = helpers.defaultNullOpts.mkBool false ''
+                  Enable showing diagnostics of this `|diagnostic-severity|` in the 'tabline'.
+                '';
+
+                icon = helpers.defaultNullOpts.mkStr iconDefault ''
+                  The icon which accompanies the number of diagnostics.
+                '';
+              };
+            in
+            {
+              "vim.diagnostic.severity.ERROR" = mkDiagnosticIconOptions " ";
+              "vim.diagnostic.severity.HINT" = mkDiagnosticIconOptions "󰌶 ";
+              "vim.diagnostic.severity.INFO" = mkDiagnosticIconOptions " ";
+              "vim.diagnostic.severity.WARN" = mkDiagnosticIconOptions " ";
+            };
+        };
+        apply = helpers.toRawKeys;
+        default = { };
+        description = helpers.defaultNullOpts.mkDesc {
+          "vim.diagnostic.severity.ERROR" = {
+            enabled = false;
+            icon = " ";
+          };
+          "vim.diagnostic.severity.HINT" = {
+            enabled = false;
+            icon = "󰌶 ";
+          };
+          "vim.diagnostic.severity.INFO" = {
+            enabled = false;
+            icon = " ";
+          };
+          "vim.diagnostic.severity.WARN" = {
+            enabled = false;
+            icon = " ";
+          };
+        } "Set the icon for each diagnostic level.";
```

Comment:
Maybe add a note to this option's description explaining that the attribute keys will be treated as raw lua key expressions?

Especially given how unusual this in nixvim.

This might be best demonstrated with an example along the lines of 

> The following config:
> ```nix
> {
>   "vim.diagnostic.severity.INFO".enabled = true;
>   "vim.diagnostic.severity.WARN".enabled = true;
> }
> ```
> Will result in the following lua:
> ```lua
> {
>   -- Note the table keys are not string literals:
>   [vim.diagnostic.severity.INFO] = { ['enabled'] = true },
>   [vim.diagnostic.severity.WARN] = { ['enabled'] = true },
> }
> ```

Reply:
Done